src/lib/import-export/export/export-manager.ts

```diff
@@ -0,0 +1,36 @@
+import fsPromises from 'fs/promises';
+import path from 'node:path';
+import { DefaultExporter, SqlExporter } from './exporters';
+import { ExportOptions, ExporterOption } from './types';
+import { WordPressExportValidator } from './validators/wordpress-validator';
+
+export async function exportBackup(
+	exportOptions: ExportOptions,
+	options: ExporterOption[] = defaultExporterOptions
+): Promise< void > {
+	const directoryContents = await fsPromises.readdir( exportOptions.sitePath, {
+		recursive: true,
+		withFileTypes: true,
+	} );
+	const allFiles = directoryContents.reduce( ( files: string[], directoryContent ) => {
+		if ( directoryContent.isFile() ) {
+			files.push( path.join( directoryContent.path, directoryContent.name ) );
+		}
+		return files;
+	}, [] as string[] );
```

Comment:
Nitpick: Here's an alternative to specify the type of output:

```suggestion
	const allFiles = directoryContents.reduce< string[] >( ( files: string[], directoryContent ) => {
		if ( directoryContent.isFile() ) {
			files.push( path.join( directoryContent.path, directoryContent.name ) );
		}
		return files;
	}, [] );
```